source/contribute/maintainer.rst

```diff
@@ -59,16 +67,167 @@ Optionally, add any additional contributor/maintainer forks, e.g.:
+`Semantic Versioning (SemVer) <https://semver.org/spec/v2.0.0.html>`_
+rules, with the famous ``MAJOR.MINOR.PATCH`` naming scheme.
+
+This effectively means:
+
+- **PATCH** only contains **bug fixes**, **security fixes** and
+  **critical fixes**
+- **MINOR** can contain everything that is in **PATCH** plus
+  **new features**, **improvements** and **deprecations**
+- **MAJOR** can contain all of the above plus **BC breaks**
+
+Whilst it is possible to automate some of these decisions, humans
+are still better at categorising these changes due to the amount of
+nuances that are involved in the software development process.
+
 Branching Model
 ---------------
 
-Merging topic branches:
+In order to maintain all the stability invariants that SemVer imposes,
+it is vital that maintainers know where to merge incoming patches.
+
+Packages in the Doctrine organisation should use the following branching
+structure:
+
+ * ``develop`` - extremely unstable, points at the next planned
+   **MAJOR** release, may be rebased in order to speed up individual
+   maintainers prototyping new changes. Changes on ``develop`` can
+   be radical, and should not be relied upon.
+ * ``master`` - always to be considered as the next planned **MAJOR**
+   or **MINOR** release (depending on team internal agreement).
+   Consumers should not rely on ``master`` unless they are prepared
+   to adapt their codebase at every potentially breaking change.
+ * ``MAJOR.MINOR.x`` - always to be considered the next planned
+   **PATCH** release. Maintainers should keep these (multiple) branches
+   stable. The base of these branches MUST be the ``MAJOR.MINOR.0`` tag.
+   Without a pre-existing tag, these branches should not exist
+
+Releasing packages
+------------------
+
+**MAJOR.0.0** and **MAJOR.MINOR.0** releases **MUST** be tagged from
+``master``.
+
+When tagging a new **MAJOR.0.0** or **MAJOR.MINOR.0** release, a
+corresponding **MAJOR.MINOR.x** branch should be branched off the tag.
+
+**MAJOR.MINOR.1+** releases must be tagged from the corresponding
+``MAJOR.MINOR.x`` branch.
+
+This effectively means that a typical Doctrine package should have a
+git graph like following:
+
+.. code-block:: console
+
+                                             ----- develop
+                                            /
+    1.0.0 ----- 1.1.0 ----- 2.0.0 ------ master
+      |           |           \
+      |           |            ----- 2.0.x
+      |           \
+      |            ----- 1.1.1 ----- 1.1.2 ----- 1.1.x
+      \
+       ----- 1.0.1 ----- 1.0.2 ----- 1.0.x
+
+Preparing a release
+-------------------
 
--  Topic branches **must** merge into **master** and/or any affected
-   release branches.
--  Merging a topic branch puts it into the *next* release, that is the
-   next release created from **master** and/or the next patch release
-   created from a specific release branch.
+Assuming that all tasks for a planned release are completed, a
```

Comment:
Given or asserting? Assuming usually leads to trouble 😄

Reply:
Please suggest an edit

Reply:
```suggestion
Given that all tasks for a planned release are completed, a
```